### PR TITLE
Filter by id on image-sets view route

### PR DESCRIPTION
**Pull request by de1987**
# Description
Add filter by id on image-sets view route

## Type of change

What is it?

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Documentation update
- [ ] Tests update
- [ ] Refactor

# Checklist:

- [ ] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] I run `make pre-commit` to check fmt/vet/lint/test-no-fdo

### CODE REVIEW

**Review by omaciel — APPROVED** [review]
/lgtm

File: pkg/routes/imagesets_test.go [hunk 1]
```diff
@@ -595,6 +595,42 @@ var _ = Describe("ImageSets Route Test", func() {
 			}
 		})
 
+		It("The image-sets view route should filter by id", func() {
```
Comment (acosferreira) [inline pkg/routes/imagesets_test.go:598]: what should happens when a invalid id was passed? Can you include this test scenario?

**Review by acosferreira — CHANGES_REQUESTED** [review]

File: pkg/routes/imagesets_test.go (hunk 1 above)
Comment (de1987) [reply pkg/routes/imagesets_test.go]: done

**Review by acosferreira — DISMISSED** [review]
lgtm

File: pkg/routes/imagesets.go [hunk 2]
```diff
@@ -353,8 +357,16 @@ func validateFilterParams(next http.Handler) http.Handler {
 			if err != nil {
 				errs = append(errs, common.ValidationError{Key: "version", Reason: fmt.Sprintf("%s is not a valid version type, version must be number", val)})
 			}
+		}
+
+		if val := r.URL.Query().Get("id"); val != "" {
+			_, err := strconv.Atoi(val)
+			if err != nil {
+				errs = append(errs, common.ValidationError{Key: "id", Reason: fmt.Sprintf("%s is not a valid id type, id must be number", val)})
```
Comment (tomershinhar) [inline pkg/routes/imagesets.go:365]: maybe it is better to say that id must be an integer?

Comment (de1987) [reply pkg/routes/imagesets.go:365]: @tomershinhar I see your point, here I'm following the pattern of the message from another validation in this function.
I noticed other pieces of the code use the word `integer` as you suggested.
Should we propose to refactor this in all edge-api code?

**Review by ldjebran — APPROVED** [review]
LGTM

File: pkg/routes/imagesets.go (hunk 2 above)
Comment (de1987) [reply pkg/routes/imagesets.go]: @tomershinhar done

**Review by tomershinhar — APPROVED** [review]
LGTM

**Review by ldjebran — APPROVED** [review]

**Comment by thom-at-redhat** [comment]
@mergifyio rebase

**Pull request merged**